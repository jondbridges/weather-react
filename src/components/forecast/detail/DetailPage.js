import React from 'react';
import Detail from './Detail';

class DetailPage extends React.Component {
  render() {
    return (
      <div>
        <Detail
          forecastDay={this.props.location.state.forecastDay}
          cityState={this.props.routeParams.cityState} />
      </div>
    );
  }
}

export default DetailPage;
