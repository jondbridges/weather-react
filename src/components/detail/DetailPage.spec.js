import React from 'react';
import renderer from 'react-test-renderer';
import DetailPage from './DetailPage';

describe('DetailPage', () => {

  it('creates the DetailPage content', () => {
    const props = {
      location: {
        state: {
          forecastDay: {
            desc: 'lots of clouds',
            maxTemp: 88,
            minTemp: 44,
            humidity: 100
          }
        }
      },
      routeParams: {
        cityState: 'Seattle'
      }
    };

    const component = renderer.create(
      <DetailPage {...props} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});