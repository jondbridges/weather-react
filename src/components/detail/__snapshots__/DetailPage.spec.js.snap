exports[`DetailPage creates the DetailPage content 1`] = `
<div>
  <div
    className="Detail-container">
    <div
      className="Forecast-day">
      <img
        alt="Weather"
        className="center-block Forecast-weather-icon"
        src="/images/weather-icons/undefined.svg" />
      <div
        className="text-center" />
    </div>
    <div
      className="col-md-4 col-md-offset-4 text-center">
      <p>
        Seattle
      </p>
      <p>
        lots of clouds
      </p>
      <p>
        <label>
          Max Temperature:
        </label>
         
        88
         °F
      </p>
      <p>
        <label>
          Min Temperature:
        </label>
         
        44
         °F
      </p>
      <p>
        <label>
          Humidity:
        </label>
         
        100
        %
      </p>
    </div>
  </div>
</div>
`;
